networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/connection/ReputationManager.java

```diff
@@ -53,28 +58,55 @@ public void reportInitiatedConnectionSuccessful(final PeerAddress peerAddress) {
     getOrCreateReputation(peerAddress).reportInitiatedConnectionSuccessful();
   }
 
+  public void reportDisconnection(
+      final PeerAddress peerAddress,
+      final Optional<DisconnectReason> reason,
+      final boolean locallyInitiated) {
+    getOrCreateReputation(peerAddress)
+        .reportDisconnection(timeProvider.getTimeInSeconds(), reason, locallyInitiated);
+  }
+
   private Reputation getOrCreateReputation(final PeerAddress peerAddress) {
-    return peerReputations.get(peerAddress, key -> new Reputation());
+    return peerReputations.get(peerAddress.getId(), key -> new Reputation());
   }
 
   private static class Reputation {
-    private static final UnsignedLong FAILURE_BAN_PERIOD = UnsignedLong.valueOf(60); // Seconds
     private volatile Optional<UnsignedLong> lastInitiationFailure = Optional.empty();
+    private volatile boolean unsuitable = false;
 
     public void reportInitiatedConnectionFailed(final UnsignedLong failureTime) {
       lastInitiationFailure = Optional.of(failureTime);
     }
 
     public boolean shouldInitiateConnection(final UnsignedLong currentTime) {
-      return lastInitiationFailure
-          .map(
-              lastFailureTime ->
-                  lastFailureTime.plus(FAILURE_BAN_PERIOD).compareTo(currentTime) < 0)
-          .orElse(true);
+      return !unsuitable
+          && lastInitiationFailure
+              .map(
+                  lastFailureTime ->
+                      lastFailureTime.plus(FAILURE_BAN_PERIOD).compareTo(currentTime) < 0)
+              .orElse(true);
     }
 
     public void reportInitiatedConnectionSuccessful() {
       lastInitiationFailure = Optional.empty();
     }
+
+    public void reportDisconnection(
+        final UnsignedLong disconnectTime,
+        final Optional<DisconnectReason> reason,
+        final boolean locallyInitiated) {
+      if (isLocallyConsideredUnsuitable(reason, locallyInitiated)
+          || reason.map(DisconnectReason::isPermanent).orElse(false)) {
```

Comment:
Isn't making a peer totally unsuitable for connection if they disconnect one time a bit harsh?

Reply:
Now I understand that we only disconnect permanently if we're unable to verify their network, or if they're on an irrelevant network. Smart.

Reply:
Yep and it only means that we don't connect to them - they can still connect to us if they want.